clients/src/main/java/org/apache/kafka/clients/consumer/KafkaConsumer.java

```diff
@@ -452,6 +453,13 @@
  *     consumer.wakeup();
  * </pre>
  *
+ * Note that while methods that can throw {@link org.apache.kafka.common.errors.WakeupException} in response to {@link #wakeup()}
+ * can also throw {@link org.apache.kafka.common.errors.InterruptException} in response to the calling thread being interrupted,
+ * it is discouraged to use thread interrupts to stop/signal a consumer thread. Preferring {@link #wakeup()} will in some cases
+ * allow for a cleaner shutdown than interrupts. {@link #commitSync()} will for example be allowed to finish committing offsets
```

Comment:
You are right, I misremembered. Updated this, and added a missing throws to `KafkaConsumer.close`